### PR TITLE
in freeSctpSession, deregister from sctp timer first

### PR DESCRIPTION
**Pull request by chehefen**
*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -100,13 +100,11 @@ matrix:
...
     - name: "Linux GCC 4.4 Build"
       os: linux
-      addons:
-        apt:
-          sources:
-            - ubuntu-toolchain-r-test
-          packages:
-            - gcc-4.4
-            - gdb
+      before_install:
```
Comment (MushMal) [inline .travis.yml:103]: This is the only line change I did for PIC:

sources:
    - sourceline: 'ppa:ubuntu-toolchain-r/test'

File: src/source/Sctp/Sctp.c [hunk 2]
```diff
@@ -135,11 +135,15 @@ STATUS freeSctpSession(PSctpSession* ppSctpSession)
 
     CHK(pSctpSession != NULL, retStatus);
 
+    usrsctp_deregister_address(pSctpSession);
```
Comment (MushMal) [inline src/source/Sctp/Sctp.c:138]: Please include some comments on what you are doing and how you model it through sctp public APIs

File: src/source/Sctp/Sctp.c [hunk 3]
```diff
@@ -135,11 +135,15 @@ STATUS freeSctpSession(PSctpSession* ppSctpSession)
...
     if (pSctpSession->socket != NULL) {
+        usrsctp_set_ulpinfo(pSctpSession->socket, NULL);
+        usrsctp_shutdown (pSctpSession->socket, SHUT_RDWR);
         usrsctp_close(pSctpSession->socket);
-        usrsctp_deregister_address(pSctpSession);
     }
 
+    THREAD_SLEEP(2 * HUNDREDS_OF_NANOS_IN_A_SECOND);
```
Comment (MushMal) [inline src/source/Sctp/Sctp.c:145]: Is this necessary?

File: src/source/Sctp/Sctp.c [hunk 4]
```diff
@@ -128,16 +129,28 @@ STATUS freeSctpSession(PSctpSession* ppSctpSession)
...
+    ATOMIC_STORE(&pSctpSession->shutdownStatus, SCTP_SESSION_SHUTDOWN_INITIATED);
+
     if (pSctpSession->socket != NULL) {
+        usrsctp_set_ulpinfo(pSctpSession->socket, NULL);
+        usrsctp_shutdown (pSctpSession->socket, SHUT_RDWR);
         usrsctp_close(pSctpSession->socket);
-        usrsctp_deregister_address(pSctpSession);
+    }
+
+    shutdownTimeout = GETTIME() + DEFAULT_SCTP_SHUTDOWN_TIMEOUT;
```
Comment (MushMal) [inline src/source/Sctp/Sctp.c:151]: Might be cleaner with a CVAR triggering but still acceptable

Comment (chehefen) [reply src/source/Sctp/Sctp.c:151]: have to introduce a lock in order to use cvar. therefore sleep instead.

**Pull request merged**